pkg/apply/task/inv_add_task_test.go

```diff
@@ -72,110 +78,164 @@ func TestInvAddTask(t *testing.T) {
-			applyObjs:    []*unstructured.Unstructured{obj2, obj3},
-			expectedObjs: object.ObjMetadataSet{id1, id2, id3},
+		"three prune objects, two same objects; three merged inventory": {
+			applyObjs:        []*unstructured.Unstructured{obj2, obj3},
+			pruneObjs:        []*unstructured.Unstructured{obj1},
+			expectedIds:      object.ObjMetadataSet{id2, id3, id1},
+			expectedSpecObjs: []actuation.ObjectReference{ref2, ref3, ref1},
+			expectedStatusObjs: []actuation.ObjectStatus{
+				{
+					ObjectReference: ref2,
+					Strategy:        actuation.ActuationStrategyApply,
+					Actuation:       actuation.ActuationPending,
+					Reconcile:       actuation.ReconcilePending,
+				},
+				{
+					ObjectReference: ref3,
+					Strategy:        actuation.ActuationStrategyApply,
+					Actuation:       actuation.ActuationPending,
+					Reconcile:       actuation.ReconcilePending,
+				},
+				{
+					ObjectReference: ref1,
+					Strategy:        actuation.ActuationStrategyDelete,
+					Actuation:       actuation.ActuationPending,
+					Reconcile:       actuation.ReconcilePending,
+				},
+			},
 		},
 	}
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			client := inventory.NewFakeClient(tc.initialObjs)
+			client := &inventory.InMemoryClient{}
 			eventChannel := make(chan event.Event)
 			resourceCache := cache.NewResourceCacheMap()
-			context := taskrunner.NewTaskContext(eventChannel, resourceCache)
+			taskContext := taskrunner.NewTaskContext(eventChannel, resourceCache)
+
+			// initialize inventory reference in context
+			inv := taskContext.InventoryManager().Inventory()
+			inv.SetGroupVersionKind(inventoryObj.GroupVersionKind())
+			inv.SetName(inventoryObj.GetName())
+			inv.SetNamespace(inventoryObj.GetNamespace())
+			inv.SetLabels(inventoryObj.GetLabels())
 
 			task := InvAddTask{
-				TaskName:  taskName,
-				InvClient: client,
-				InvInfo:   nil,
-				Objects:   tc.applyObjs,
-			}
-			if taskName != task.Name() {
-				t.Errorf("expected task name (%s), got (%s)", taskName, task.Name())
+				TaskName:     taskName,
+				InvClient:    client,
+				ApplyObjects: tc.applyObjs,
+				PruneObjects: tc.pruneObjs,
 			}
-			applyIds := object.UnstructuredSetToObjMetadataSet(tc.applyObjs)
-			if !task.Identifiers().Equal(applyIds) {
-				t.Errorf("expected task ids (%s), got (%s)", applyIds, task.Identifiers())
-			}
-			task.Start(context)
-			result := <-context.TaskChannel()
-			if result.Err != nil {
-				t.Errorf("unexpected error running InvAddTask: %s", result.Err)
-			}
-			actual, _ := client.GetClusterObjs(nil)
-			if !tc.expectedObjs.Equal(actual) {
-				t.Errorf("expected merged inventory (%s), got (%s)", tc.expectedObjs, actual)
+			assert.Equal(t, taskName, task.Name())
+			testutil.AssertEqual(t, tc.expectedIds, task.Identifiers())
+
+			task.Start(taskContext)
+			result := <-taskContext.TaskChannel()
+			if tc.expectedError != nil {
+				assert.EqualError(t, result.Err, tc.expectedError.Error())
```

Comment:
Here's another assert.EqualError cleanup that could be bundled with others in another PR.